src/charm.py

```diff
@@ -407,6 +410,15 @@ def set_secret(self, scope: str, key: str, value: Optional[str]) -> None:
         else:
             raise RuntimeError("Unknown secret scope.")
 
+    def install_exporter(self) -> None:
+        """Install JMX exporter."""
+        url = "https://repo1.maven.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/0.17.2/jmx_prometheus_javaagent-0.17.2.jar"
```

Comment:
This isn't a great move; if you rely upon external internet access then any airgapped deployments will break. This needs to be either shipped as part of the snap, or attached to the charm as a resource and published with it :)